packages/openapi-typescript/src/load.ts

```diff
@@ -279,6 +279,10 @@ export default async function load(
 
         const ref = parseRef(node.$ref);
 
+        // when ref.path is [], $ref has been transform, skip transform
+        if (ref.path.length === 0 && (ref.filename.startsWith("components[") || ref.filename.startsWith("external["))) {
```

Comment:
Hm this may not be true in all cases; `$ref`s could also reference the `operations` interface as well.

This feels like masking a potential bug rather than pinpointing exactly _why_ the `$ref` isn’t being transformed correctly.

Is this an invalid schema?

Or is this a specific bug with specific `$ref`s?

Rather than change this, we should find out _exactly_ what kind of `$ref` isn’t being transformed properly.

Reply:
The reason for this bug is: 
use items: &ref_0 and items: *ref_0, Causes the ref reference in the currentSchema to point to the same address, and the currentSchema data has changed, resulting in ref transform many times！
I use JSON.parse and JSON.stringify to handle schema data before walk currentSchema to refactor code.